packages/flutter_tools/lib/src/base/config.dart

```diff
@@ -3,15 +3,23 @@
 // found in the LICENSE file.
 
 import '../convert.dart';
+import '../globals.dart';
 import 'context.dart';
 import 'file_system.dart';
+import 'logger.dart';
 import 'utils.dart';
 
 class Config {
-  Config([File configFile]) {
+  Config([File configFile, Logger localLogger]) {
+    final Logger loggerInstance = localLogger ?? logger;
     _configFile = configFile ?? fs.file(fs.path.join(userHomePath(), '.flutter_settings'));
     if (_configFile.existsSync()) {
-      _values = castStringKeyedMap(json.decode(_configFile.readAsStringSync()));
+      try {
+        _values = castStringKeyedMap(json.decode(_configFile.readAsStringSync()));
+      } on FormatException {
+        loggerInstance.printError('Failed to decode preferences in ${_configFile.path}.');
+        _configFile.deleteSync();
```

Comment:
Should we delete this (I'm guessing probably, but wanted to check)? Are we assuming that the tool messed up, or are users manually editing these?

Assuming these are pretty small, should we at least log the contents of the corrupted config file?

Reply:
In the filled issue it looks like the file was corrupted - we could log the contents, but I'm worried about an edge case where it is accidentally a 2GB file.

Reply:
that's a good point about the edge case...

Reply:
Do we persist analytics opt in data here?

Reply:
The analytics opt in data is stored in $HOME/.flutter

Reply:
I guess it's ok to blow this file away, then.

Reply:
I added more to the messaging regarding the deletion